cmd/tailscale/cli/file.go

```diff
@@ -413,52 +415,70 @@ func runFileGet(ctx context.Context, args []string) error {
 	if fi, err := os.Stat(dir); err != nil || !fi.IsDir() {
 		return fmt.Errorf("%q is not a directory", dir)
 	}
-
-	var wfs []apitype.WaitingFile
-	var err error
-	for {
-		wfs, err = tailscale.WaitingFiles(ctx)
-		if err != nil {
-			return fmt.Errorf("getting WaitingFiles: %w", err)
+	for true {  // Infinite loop
+		var wfs []apitype.WaitingFile
+		var err error
+		var errs []error
+		for {
+			wfs, err = tailscale.WaitingFiles(ctx)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("getting WaitingFiles: %w", err))
+			}
+			if len(wfs) != 0 || !(getArgs.wait || getArgs.loop) {
+				break
+			}
+			if getArgs.verbose {
+				printf("waiting for file...")
+			}
+			if err := waitForFile(ctx); err != nil {
+				errs = append(errs, err)
+			}
 		}
-		if len(wfs) != 0 || !getArgs.wait {
-			break
+
+		deleted := 0
+		for _, wf := range wfs {
+			writtenFile, size, err := receiveFile(ctx, wf, dir)
+			if err != nil {
+				errs = append(errs, err)
+				continue
+			}
+			if getArgs.verbose {
+				printf("wrote %v as %v (%d bytes)\n", wf.Name, writtenFile, size)
+			}
+			if err = tailscale.DeleteWaitingFile(ctx, wf.Name); err != nil {
+				errs = append(errs, fmt.Errorf("deleting %q from inbox: %v", wf.Name, err))
+				continue
+			}
+			deleted++
 		}
 		if getArgs.verbose {
-			printf("waiting for file...")
-		}
-		if err := waitForFile(ctx); err != nil {
-			return err
+			printf("moved %d/%d files\n", deleted, len(wfs))
 		}
-	}
-
-	var errs []error
-	deleted := 0
-	for _, wf := range wfs {
-		writtenFile, size, err := receiveFile(ctx, wf, dir)
-		if err != nil {
-			errs = append(errs, err)
-			continue
+		if !getArgs.loop {
+			if len(errs) == 0 {
+				return nil
+			}
+			for _, err := range errs[:len(errs)-1] {
+				outln(err)
+			}
+			return errs[len(errs)-1]
 		}
-		if getArgs.verbose {
-			printf("wrote %v as %v (%d bytes)\n", wf.Name, writtenFile, size)
+		// deamon mode
+		for _, err := range errs {
+			outln(err)
 		}
-		if err = tailscale.DeleteWaitingFile(ctx, wf.Name); err != nil {
-			errs = append(errs, fmt.Errorf("deleting %q from inbox: %v", wf.Name, err))
-			continue
+		if len(errs) > 0 {
+			// It's possible whatever caused the error(s) (e.g. conflicting target file,
+			// full disk, unwritable target directory) will re-occur if we try again so
+			// let's back off and not busy loop on error.
+			//
+			// However, let's not wait too long so other files may get received if
+			// there's just one problematic file.
+			time.Sleep(5 * time.Second)
```

Comment:
we don't always want this sleep. only in the loop case.

but I think you actually want to use `tailscale.com/logtail/backoff.BackOff`

Reply:
the sleep only got executed in the loopy case on error, but the structure was making that really non-obvious.  (the "non-loop" case had returns in all of its branches).

So the failure scenario that had me shove a 5 second sleep in there is someone running:
  `tailscale file get --conflict=skip --loop`

Say tailscaled has received an `image.jpg` along with a bunch of PDFs the user is trying to save for later, and we already have an `image.jpg` in our Downloads directory.  `tailscale file get` will record an error along the lines of "hey, so I couldn't write to Downloads/image.jpg, because you told me to "skip" instead of "overwrite", and you've already got one there, so that file is wedged in your inbox."  it's a persistent error that will happen every time but is sort of ok.  we're happy to receive *other* files, just not that one.  and we don't want to start backing off and waiting ten minutes to receive any file because there's one wedged file.  on the other hand, if we ask tailscaled to tell us when it's got files, hey, it's got files! there's this `image.jpg` it keeps telling us about.